plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Future.hs

```diff
@@ -317,10 +319,18 @@ machineClient inst future ftos =
 validator :: Future -> FutureAccounts -> Validator
 validator ft fos = Scripts.validatorScript (scriptInstance ft fos)
 
-{-# INLINABLE verifyOracle #-}
-verifyOracle :: Future -> OracleValue a -> Maybe (Slot, a)
-verifyOracle Future{ftPriceOracle} OracleValue{ovSignature, ovSlot, ovValue} =
-    if ovSignature == ftPriceOracle then Just (ovSlot, ovValue) else Nothing
+{-# INLINABLE verifyOracleOnChain #-}
+verifyOracleOnChain :: PlutusTx.IsData a => PendingTx -> Future -> SignedMessage (Observation a) -> Maybe (Slot, a)
+verifyOracleOnChain ptx Future{ftPriceOracle} sm@SignedMessage{osmSignature, osmMessageHash, osmData} =
+    case Oracle.checkSignature osmMessageHash ftPriceOracle osmSignature >> Oracle.checkHashOnChain ptx sm of
```

Comment:
Seems like we need a function that's the composition of these two, should be fine since they're both `Either`s with the same error type.